addon/corpus.js

```diff
@@ -1,106 +1,231 @@
-
-    let namesAndUrls;
-    if (lines[0].includes(' ') || lines[0].includes('\t')) {
-        // We have explicit filename prepended to the lines, space-delimited.
-        namesAndUrls = lines.map(function splitAndSuffix(l) {
-            let [name, url] = l.split(/[ \t]+/, 2);
-            return [name + '.html', url];
-        });
-    } else {
-        namesAndUrls = lines.map(l => [undefined, l]);
+    windowId = (await browser.windows.create({
+        url: '/pages/blank.html',
+        width: 1024,
+        height: 768,
+    })).id;
+}
+document.getElementById('freeze').onclick = freezeAllPages;
+
+function fathomNext(event) {
+    const windowId = event.detail;
+
+    // Load next tab, or close the window if we're done.
+    gUrlIndex++;
+    if (gUrlIndex >= gUrls.length) {
+        browser.windows.remove(windowId);
+        document.getElementById('freeze').disabled = false;
+        return;
     }
 
-    const freezeOptions = {wait: parseFloat(document.getElementById('wait').value.trim()),
-                           shouldScroll: document.getElementById('shouldScroll').checked};
-    for (let [filename, url] of namesAndUrls) {
-        try {
-            await freezePage(url, windowId, freezeOptions, filename);
-        } catch (e) {
-            // What can go wrong? Redirects mess up our messaging pipeline.
-            // Modal alerts hang us until the user dismisses them.
-            errorField.appendChild(document.createTextNode(`\nError while freezing ${url}: ${e}`));
+    // Create a new tab with the current url, always as tab[0].
+    // The tabs.onUpdated handler in freezeAllPages() will dispatch a fathom:freeze
+    // event when the tab has completed loading.
+    setCurrentStatus('loading');
+    browser.tabs.create({
+        windowId: windowId,
+        index: 0,
+        url: gUrls[gUrlIndex].url,
+    });
+}
+document.addEventListener('fathom:next', fathomNext, false);
+
+async function fathomFreeze(event) {
+    const windowId = event.detail.windowId;
+    const timer = event.detail.timer;
+    const tab = (await browser.tabs.query({ windowId }))[0];
+
+    setCurrentStatus('freezing');
+    try {
+        // Can't get a return value out of the content script because webpack wraps
+        // our top-level stuff in a function. Instead, we use messaging.
+        await browser.tabs.executeScript(
+            tab.id,
+            { file: '/contentScript.js' }
+        );
+
+        // Call freeze-dry to fetch html.
+        const html = await browser.tabs.sendMessage(
+            tab.id,
+            { type: 'freeze', options: gFreezeOptions }
+        );
+
+        // Clear timeout here so we don't bail out while writing to disk.
+        clearTimeout(timer);
+
+        // Save html to disk.
+        const filename = gUrls[gUrlIndex].filename;
+        await download(html, { filename });
+
+        setCurrentStatus('downloaded');
+    } catch(e) {
```

Comment:
The other place we use `catch`, we put a space before (.

Reply:
adding `keyword-spacing` to `.eslintrc`